adal/src/main/java/com/microsoft/aad/adal/AcquireTokenSilentHandler.java

```diff
@@ -118,6 +118,78 @@ AuthenticationResult getAccessToken() throws AuthenticationException {
         return AuthenticationResult.createResult(accessTokenItem);
     }
 
+    AuthenticationResult getAccessTokenUsingAssertion() throws AuthenticationException {
+        final String methodName = ":getAccessTokenUsingAssertion";
+        final AuthenticationResult result = acquireTokenWithAssertion();
+        
+        if (result != null && !StringExtensions.isNullOrBlank(result.getAccessToken())) {
+            try {
+                mTokenCacheAccessor.updateTokenCache(mAuthRequest, result);
+            } catch (MalformedURLException e) {
+                Logger.v(TAG + methodName, "Access token fetched but unable to update token cache");
+                throw new AuthenticationException(ADALError.DEVELOPER_AUTHORITY_IS_NOT_VALID_URL, e.getMessage(), e);
+            }
+        }
+
+        return result;
+    }
+
+    /**
+     * Send token request with grant_type as assertion to token endpoint for getting new access token.
+     */
+    AuthenticationResult acquireTokenWithAssertion() throws AuthenticationException {
+        final String methodName = ":acquireTokenWithAssertion";
+        Logger.v(TAG + methodName, "Try to get new access token with the provided assertion.",
+                mAuthRequest.getLogInfo(), null);
+
+        // Check if network is available, if not throw exception. 
+        HttpUtil.throwIfNetworkNotAvailable(mContext);
+
+        final AuthenticationResult result;
+        final String samlAssertion = mAuthRequest.getSamlAssertion();
+        final String assertionType = mAuthRequest.getAssertionType();
+
+        try {
+            final JWSBuilder jwsBuilder = new JWSBuilder();
+            final Oauth2 oauthRequest = new Oauth2(mAuthRequest, mWebRequestHandler, jwsBuilder);
+
+            result = oauthRequest.refreshTokenUsingAssertion(samlAssertion, assertionType);
+            if (result != null && StringExtensions.isNullOrBlank(result.getRefreshToken())) {
+                Logger.i(TAG + methodName, "Refresh token is not returned or empty", "");
```

Comment:
This seems more like a `warning` than an `info` statement, right?

Reply:
Yes, that's right. Will change it

Reply:
> This seems more like a `warning` than an `info` statement, right?

The corresponding line in the refreshtoken method at line 215 is also logging as an info, do you want me to change that too?

Reply:
Yes please 👍